### PR TITLE
Initial UI for SignUp and SignIn

### PR DESCRIPTION
**Pull request by hobi9**
I added an initial implementation for the SignUp and SignIn components. At the moment they are not connected to the backend so there's no navigation in case of a successful login/registration. Also there is no username validation (green outline in case the username is valid/unique) and eventual server errors.

### CODE REVIEW

**Review by MrBisquit — APPROVED** [review]
Looks all good to me!

**Comment by KingXP-Pythoner** [comment]
Your PR doesn't utilize the shadcn Input component which contains the styling.

**Comment by hobi9** [comment]
> Your PR doesn't utilize the shadcn Input component which contains the styling. An overflow scrollbar on the x axis pops up on validation error.

i fixed the overflow and imported the input component. Also added zod

**Review by MrBisquit — APPROVED** [review]
The update looks good.

File: frontend/src/components/ui/DiscordLogo.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const { theme } = useTheme();
+
+  return (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      viewBox="0 0 127.14 96.36"
+      height={size}
+      width={size}
+    >
+      <g id="图层_2" data-name="图层 2">
```
Comment (MrBisquit) [inline frontend/src/components/ui/DiscordLogo.tsx:16]: Why is it 图层 rather than layer?

**Comment by MrBisquit** [comment]
I hate miss clicking, just started a review rather than making a comment.

File: frontend/src/components/ui/DiscordLogo.tsx (hunk 1 above)
Comment (hobi9) [reply frontend/src/components/ui/DiscordLogo.tsx:16]: That's how I got it from the official discord site

**Comment by hobi9** [comment]
> Need to remove '^' from package.json after installing packages. Also, there is an update needed which is Full Name field. This is coming from backend and wasnt updated in the figma design yet.
> 
> 

Thanks for the reply. I'll remove the carrot and open a new pr once I see the updated figma file.

File: frontend/src/components/ui/DiscordLogo.tsx (hunk 1 above)
Comment (MrBisquit) [reply frontend/src/components/ui/DiscordLogo.tsx:16]: Alright

**Review by MrBisquit — APPROVED** [review]

**Comment by KingXP-Pythoner** [comment]
can you make it so that when the input error ring is activated, it actually changes to the success color when the user types and meets the requirement. Because if it's still red while they're typing, there's no feedback that it's correct and can start wondering why it's still not correct. There should be aria-invalid attribute added as well to help with accessibility reasons

**Comment by hobi9** [comment]
> can you make it so that when the input error ring is activated, it actually changes to the success color when the user types and meets the requirement. Because if it's still red while they're typing, there's no feedback that it's correct and can start wondering why it's still not correct. There should be aria-invalid attribute added as well to help with accessibility reasons

Done. I fixed it by adding react form which made it easier since the form became bigger. Also added those accessibility attributes.

**Comment by SHIV5T3R** [comment]
Please fix conflicts, yarn.lock has already been removed in upstream.

**Pull request closed without merging**